### PR TITLE
refactor: light program

### PR DESCRIPTION
**Pull request by ananas-block**
builds on https://github.com/Lightprotocol/light-protocol/pull/480

changes:
- light program
  - added instructions and todo comments for instructions
  - adapted tests
- test-utils
  - added init test env function

### CODE REVIEW

File: programs/light/tests/tests.rs [hunk 1]
```diff
@@ -1,179 +1,97 @@
...
-    program_test.set_compute_max_units(1_400_000u64);
-    let authority_pda = Pubkey::find_program_address(
-        &[light::AUTHORITY_PDA_SEED, &light::ID.to_bytes().as_slice()],
-        &light::ID,
-    );
-    let mut context = program_test.start_with_context().await;
-    let updated_keypair = Keypair::new();
-
-    airdrop_lamports(&mut context, &updated_keypair.pubkey(), 1_000_000_000)
+pub const LIGHT_ID: Pubkey = pubkey!("5WzvRtu7LABotw1SUEpguJiKU27LRGsiCnF5FH6VV7yP");
```
Comment (vadorovsky) [inline programs/light/tests/tests.rs:17]: You can probably import it from test-utils, same with the other `pubkey!` constants

File: utils/src/lib.rs [hunk 2]
```diff
@@ -87,3 +119,26 @@ pub fn rustfmt(code: String) -> Result<Vec<u8>, anyhow::Error> {
...
+        for _ in 0..1000_0000 {
+            let input_bytes = Pubkey::new_unique().to_bytes(); // Sample input
+            if let Some((hashed_value, _)) = hash_to_bn254_field_size_le(input_bytes.as_slice()) {
+                assert!(
+                    is_smaller_than_bn254_field_size_le(&hashed_value).unwrap(),
+                    "Hashed value should be within BN254 field size"
+                );
+            } else {
+                panic!("Failed to find a hash within BN254 field size");
+            }
```
Comment (vadorovsky) [inline utils/src/lib.rs:141]:
```suggestion
            let (hashed_value, _) = hash_to_bn254_field_size_le(input_bytes.as_slice()).expect("Failed to find a hash within BN254 field size");
            assert!(
                is_smaller_than_bn254_field_size_le(&hashed_value).unwrap(),
                "Hashed value should be within BN254 field size"
            );
```

File: utils/src/lib.rs [hunk 3]
```diff
@@ -57,6 +60,35 @@ pub fn truncate_to_circuit(bytes: &[u8; 32]) -> [u8; 32] {
...
+            let mut hashed_value: [u8; 32] = hashv(&[bytes, bump_seed.as_ref()]).to_bytes();
+            // TODO: revisit truncation
+            // truncate to 31 bytes so that value is less than bn254 Fr modulo field
+            hashed_value[31] = 0;
+
+            match is_smaller_than_bn254_field_size_le(&hashed_value) {
+                Ok(true) => return Some((hashed_value, bump_seed[0])),
+                Ok(false) => (),
+                Err(_) => (),
+            }
```
[the remark this answers is not in the document]
Comment (ananas-block) [reply utils/src/lib.rs:85]: fixed in other pr

File: utils/src/lib.rs (hunk 2 above)
Comment (ananas-block) [reply utils/src/lib.rs:141]: fixed in different pr

File: programs/light/tests/tests.rs (hunk 1 above)
Comment (ananas-block) [reply programs/light/tests/tests.rs:17]: actually I can just take it from it's own crate light::ID;

**Review by vadorovsky — APPROVED** [review]

**Pull request merged**